### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'end'

### ISSUE THREAD

**Issue by Marenz** [simplistix/sybil#50]
I am not sure if it is due to me using it wrong or simply a missing check, but i am getting this error:

```
.direnv/python-3.11.2/lib64/python3.11/site-packages/sybil/integration/pytest.py:104: in collect
    self.document = self.sybil.parse(Path(self.fspath.strpath))
.direnv/python-3.11.2/lib64/python3.11/site-packages/sybil/sybil.py:144: in parse
    return type_.parse(str(path), *self.parsers, encoding=self.encoding)
.direnv/python-3.11.2/lib64/python3.11/site-packages/sybil/document.py:196: in parse
    for start, end, text in cls.extract_docstrings(document.text):
.direnv/python-3.11.2/lib64/python3.11/site-packages/sybil/document.py:182: in extract_docstrings
    punc_size = punc.end() - punc.start()
E   AttributeError: 'NoneType' object has no attribute 'end'
```

While investigating, it seems to be that the result of the regex search is never checked before accessing it:

https://github.com/simplistix/sybil/blob/master/sybil/document.py#L183

Thus it should fail for any file that doesn't have the expected documentation type.

**Comment by cjw296** [comment]
Are you able to share you Sybil config along with the file that causes this exception?

**Comment by Marenz** [comment]
This is how Sybil is used/started: http://github.com/frequenz-floss/frequenz-sdk-python/pull/364/commits/1034637df25dcaeb0ce813a8f92627562501fc7c

And this is the file causing problems: https://github.com/frequenz-floss/frequenz-sdk-python/blob/e4356e294fc300239b95eade0e61dc582ea38aa9/src/frequenz/sdk/power/_distribution_algorithm.py

Here is the output of a CI run that is failing: https://github.com/frequenz-floss/frequenz-sdk-python/actions/runs/4893485711/jobs/8736504619?pr=364#step:6:74

**Comment by cjw296** [comment]
Curious why the docstring on line 43 is r-prefixed? https://github.com/frequenz-floss/frequenz-sdk-python/blob/e4356e294fc300239b95eade0e61dc582ea38aa9/src/frequenz/sdk/power/_distribution_algorithm.py#L43

It's a bug in Sybil that this doesn't "just work", but the r-string seems unnecessary here.

**Comment by cjw296** [comment]
Anyway, fixed in 5.0.1, thanks for the report!

**Comment by Marenz** [comment]
The r prefix seems to be a typo/bug on our end, thanks for pointing it out!
